src/Sylius/Behat/Context/Ui/Admin/ManagingTaxonsContext.php

```diff
@@ -408,68 +374,41 @@ public function iRemoveTheFirstImage()
     /**
      * @Then /^(this taxon) should not have any images$/
      */
-    public function thisTaxonShouldNotHaveImages(TaxonInterface $taxon)
+    public function thisTaxonShouldNotHaveAnyImages(TaxonInterface $taxon)
     {
         $this->iWantToModifyATaxon($taxon);
 
         Assert::same($this->updatePage->countImages(), 0);
     }
 
     /**
-     * @When I change the image with the :code code to :path
-     */
-    public function iChangeItsImageToPathForTheCode($path, $code)
-    {
-        $this->updatePage->changeImageWithCode($code, $path);
-    }
-
-    /**
-     * @Then I should be notified that the image with this code already exists
-     */
-    public function iShouldBeNotifiedThatTheImageWithThisCodeAlreadyExists()
-    {
-        Assert::same($this->updatePage->getValidationMessageForImage(), 'Image code must be unique within this taxon.');
-    }
-
-    /**
-     * @Then I should be notified that an image code is required
+     * @When I change the image with the :type type to :path
```

Comment:
There's no way of differentiating between multiple images with the same type other than their position on the site. Filenames get cached and you end up with a beautifully mashed string.
For the scenarios with unique-typed images it's still a valid step, as for dupes, using `first` is sufficient since it tests what needs to be tested, and doesn't confound the logic.